### PR TITLE
feat: Draft goals

### LINKED ISSUE #3: Draft goals
Document the goals of the conformance program and start gathering important ideas that will need to be part of the standard. 

Inspiration:

- Prometheus https://github.com/cncf/prometheus-conformance 
- Kubernetes https://www.cncf.io/certification/software-conformance/ 
- Crossplane https://github.com/cncf/crossplane-conformance 
- https://github.com/argoproj/argo-conformance-program/blob/main/README.md

### PR DESCRIPTION
**Pull request by todaywasawesome**
Closes #3, this draft is considered a work in progress. Next steps are
[x] Committee Review
[x] Solicit Maintainer Feedback
[x] Merge

When providing feedback, please select the specific lines you'd like to address and even feel free to make edits using Github's suggestion tool. We will discuss these in the next conformance committee meeting. 

For context, consider reviewing other successful CNCF conformance programs:
* https://www.cncf.io/certification/software-conformance/
* https://github.com/cncf/crossplane-conformance
* https://github.com/cncf/prometheus-conformance

Vendors are already building forks of Argo meant to compete with the open-source offering and community. This underlines the value of building this kind of program and encouraging more community cohesion. 

Co-authored-by: Jann Fischer <jfischer@redhat.com>
Co-authored-by: Henrik Blixt <henrik_blixt@intuit.com>
Co-authored-by: Hong Wang <wanghong230@gmail.com>

### CODE REVIEW

File: README.md
Comment (todaywasawesome) [inline README.md]:
```suggestion
* Ensure a consistent and reliable user experience and API compatibility 
* Protect the Argo brand
```

From Henrik

File: README.md
Comment (todaywasawesome) [inline README.md]:
```suggestion
* Ensure contributions from commercial partners
* To grow and strengthen the Argo community
```

We will likely require active maintainers in order to meet conformance. It's not enough to encourage contributions, they should be required.

File: README.md
Comment (todaywasawesome) [inline README.md:21]:
```suggestion
* Advertise compliance and certification
* Use Argo logo and branding in marketing materials
```

Make sure that brand usage is explicitly stated.

File: README.md
Comment (todaywasawesome) [inline README.md]:
```suggestion
* Exposure to community
```

Simplify language and avoid "preference".

File: README.md
Comment (todaywasawesome) [inline README.md]:
```suggestion
* Guarantee a consistent and reliable user experience and API compatibility 
* Protect the Argo brand
* Ensure contributions from commercial partners
* Grow and strengthen the Argo community
```
Make lines consistent in form and don't reuse "ensure" too much.

File: README.md
Comment (wanghong230) [inline README.md]: What is the reason to change to Promote?

Comment (edlee2121) [reply README.md]: Is Encourage better?

Comment (todaywasawesome) [reply README.md]: @edlee2121 for any additional changes please use the code suggest feature in a code review. Which will allow people to discuss the change before merging. 

We actually started with "Encourage" and the group felt that it wasn't strong enough as a goal and that we should basically require contributions to be coming back in some way. It's likely this will take the form of certified offerings to have a maintainer in the Argoproj. 

Given that I think we should go back to "Ensure".

Comment (todaywasawesome) [reply README.md:14]:
```suggestion
* Ensure contributions from commercial partners
```

Comment (wanghong230) [reply README.md]: I like "Ensure" as the goal to really benefit the community in a measurable way. Thanks.

Comment (edlee2121) [reply README.md]: Sure not sure how to use that feature. Let me revert my commit and resubmit it.

File: README.md
Comment (edlee2121) [inline README.md]:
```suggestion
+The Argo community continues to grow dramatically – adding new users, new contributors, and new capabilities. The Argo Conformance Program creates incentives and requirements designed to help the ecosystem grow in a sustainable and user-friendly way. As a diverse set of tools grows around Argo, users can trust that products certified as compliant will interoperate with each other and be up-to-date with the latest open source releases. This ensures a consistent user experience between products and flexibility to choose between vendors.
```

File: README.md
Comment (edlee2121) [inline README.md]:
```suggestion
+* Guarantee a consistent and predictable user experience
+* Guarantee interoperability, including API compatibility
+* Guarantee conformant products are up-to-date with latest Argo releases 
+* Ensure contributions from commercial partners
 * Grow and strengthen the Argo community
+* Protect the Argo brand

```

File: README.md
Comment (todaywasawesome) [reply README.md]: LGTM 👍

File: README.md
Comment (todaywasawesome) [reply README.md]: Good additions and the wording is solid. 👍

**Review by jannfis — APPROVED** [review]
LGTM

File: README.md
Comment (edlee2121) [inline README.md]:
```suggestion
* Exposure to community

# Benefits to Community
* Promote interoperability and compatibility between Argo products
* Discourage forking of the Argo code base
* Encourage commercial interests to work together to grow the Argo ecosystem rather than purely on competition with each other
```

Comment (todaywasawesome) [reply README.md]: I like it!

File: README.md
Comment (hblixt) [inline README.md:14]: Do we want to use something softer, like "engagement" over contributions? As a goal, engagement feels better, then we can specify in the requirements exactly what it means

Comment (todaywasawesome) [reply README.md:14]: I'm good with that @hblixt. @edlee2121 made a good point when I talked to him today about a race for people to get maintainers in. I think we can work it out in the requirements but I'm fine with either word.

File: README.md
Comment (hblixt) [inline README.md]: We should leave out the negative part of the statement (ie "... rather than purely on competition with each other"). 
Focus on the positive :)

File: README.md
Comment (hblixt) [reply README.md:14]: I don't really feel strongly either way. It just felt a little more appropriate for a goal to use engagement over contributions, which seemed a bit more of a requirement.

Comment (todaywasawesome) [reply README.md:14]: Debated this in committee with @alexec @wanghong230 and @hblixt decided to move forward with current language. May revisit if someone wants conformance without maintainers.

File: README.md
Comment (todaywasawesome) [reply README.md]:
```suggestion
* Encourage commercial interests to work together to grow the Argo ecosystem
```

Comment (todaywasawesome) [reply README.md]: Discussed in committee with @hblixt @alexec and @wanghong230 agreed to remove the last part as it's covered in the first part.

**Review by wanghong230 — APPROVED** [review]

**Review by hblixt — APPROVED** [review]

**Pull request merged**
